compiler/compile/OMRCompilation.cpp

```diff
@@ -2173,6 +2173,81 @@ OMR::Compilation::isInlinedDirectCall(uint32_t index)
    return _inlinedCallSites[index].directCall();
    }
 
+/*
+ * Get the size of the OSRCallSiteRemat table, based on the number of pending push
+ * slots in the caller.
+ */
+uint32_t
+OMR::Compilation::getOSRCallSiteRematSize(uint32_t callSiteIndex)
+   {
+   if (!_inlinedCallSites[callSiteIndex].osrCallSiteRematTable())
+      return 0;
+
+   int32_t callerIndex = self()->getInlinedCallSite(callSiteIndex)._byteCodeInfo.getCallerIndex();
+   return callerIndex < 0 ? self()->getMethodSymbol()->getNumPPSlots() :
+      self()->getInlinedResolvedMethodSymbol(callerIndex)->getNumPPSlots();
+   }
+
+/*
+ * Get the pending push symbol reference and the corresponding load, to later remat the pending push 
+ * within OSR code blocks inside the callee. To get a mapping, the call site index for the callee and
+ * the caller's pending push slot should be provided.
+ */
+void
+OMR::Compilation::getOSRCallSiteRemat(uint32_t callSiteIndex, uint32_t slot, TR::SymbolReference *&ppSymRef, TR::SymbolReference *&loadSymRef)
+   {
+   int32_t *table = _inlinedCallSites[callSiteIndex].osrCallSiteRematTable();
+   if (!table)
+      {
+      ppSymRef = NULL;
+      loadSymRef = NULL;
+      return;
+      }
+
+#if defined(DEBUG) || defined(PROD_WITH_ASSUMES)
+   // Ensure the requested slot is valid, based on the total number of PP slots
+   uint32_t callerNumPPSlots = self()->getOSRCallSiteRematSize(callSiteIndex);
+   TR_ASSERT(slot < callerNumPPSlots, "can only perform call site remat for the caller's pending pushes");
+#endif
+
+   TR::SymbolReferenceTable *symRefTab = self()->getSymRefTab();
+   ppSymRef = table[slot * 2] == 0 ? NULL : symRefTab->getSymRef(table[slot * 2]);
```

Comment:
Pending push slots are shared, so it would be necessary to work out which symbol reference is within the slot at the call site. It would be possible to do with the OSR data, as it stores this information.